Nodejs/Product/TestAdapter/TestFrameworks/Jest/jest.js

```diff
@@ -110,6 +118,45 @@ function detectPackage(projectFolder, packageName) {
     }
 }
 
+function readConfigs(projectFolder, context)
+{
+    let config;
+    const packageJsonPath = path.join(projectFolder, 'package.json');
+    const jestConfigPath = path.join(projectFolder, 'jest.config.js');
+
+    // If this is a React project, then the path exists, and we should use the react scripts config generation. 
+    // It already deals with override from package.json config, if there's any.
+    const pathToReactConfig = path.join(projectFolder, 'node_modules/react-scripts/scripts/utils/createJestConfig.js');
+    if(fs.existsSync(pathToReactConfig))
+    {
+        const createJestConfig = require(pathToReactConfig);
+        config = createJestConfig(
+            relativePath => path.join(projectFolder, 'node_modules/react-scripts/', relativePath),
+            projectFolder,
+            false
+        )
+
+        return config;
+    }
+
+    // Else, for other frameworks, first prioritize package.json config if jest config exists there under "jest".
+    // If not, try to find jest.config.js. If no file found, finally use a basic config.
+    if(fs.existsSync(packageJsonPath))
+    {
+        config = require(packageJsonPath).jest;
+    }
+
+    if(!config && fs.existsSync(jestConfigPath))
+    {
+        config = require(jestConfigPath);
+    }
+
+    if(!config) config = {};
```

Comment:
nit: You can use the logical OR assignment (`||=`). In node is supported since version 15.14.0, although I don't remember which version of Node we support.